### PR TITLE
Python: Added basic structure

### PR DESCRIPTION
**Pull request by ahcorde**
# 🎉 New feature

This PR is part of this mete ticket https://github.com/ignitionrobotics/sdformat/issues/931

## Summary

This PR add the basic structure to create Python binding using pybind11

## Test it

Compile the code, export the pythonpath and run the example

```bash
export PYTHONPATH=$PYTHONPATH:<path to ws>/install/lib/python
python3 examples/python/python_sdformat.py
```

## Checklist
- [x] Signed all commits for DCO
- [ ] Added tests
- [x] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [x] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [x] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [x] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -128,6 +128,40 @@ if (BUILD_SDF)
...
+# Python interfaces
+option(USE_SYSTEM_PATHS_FOR_PYTHON_INSTALLATION
+      "Install python modules in standard system paths in the system"
+      OFF)
+
+option(USE_DIST_PACKAGES_FOR_PYTHON
+      "Use dist-packages instead of site-package to install python modules"
+      OFF)
+
+include(IgnPython)
```
Comment (scpeters) [inline CMakeLists.txt:141]: there is already some code to find python up at lines 60-90; we should unify that code with this

Comment (ahcorde) [reply CMakeLists.txt]: https://github.com/ignitionrobotics/sdformat/pull/932/commits/05545f844df895c4c1157e773061a4e5dcdb8148

File: CMakeLists.txt [hunk 2]
```diff
@@ -114,6 +124,27 @@ if (BUILD_SDF)
   ign_find_package(ignition-common5 COMPONENTS graphics REQUIRED_BY usd)
   set(IGN_COMMON_VER ${ignition-common5_VERSION_MAJOR})
 
+  ########################################
+  # Python interfaces
+  find_package(PythonLibs QUIET)
+  if (NOT PYTHONLIBS_FOUND)
+    IGN_BUILD_WARNING("Python is missing: Python interfaces are disabled.")
```
Comment (azeey) [inline CMakeLists.txt:131]: CI is not finding python
```
-- Searching for Python - not found.
-- Looking for pxr - not found

CMake Warning at /usr/share/cmake/ignition-cmake2/cmake2/IgnConfigureBuild.cmake:55 (message):
   CONFIGURATION WARNINGS:
   -- Python psutil package not found. Memory leak tests will be skipped
   -- Python is missing: Python interfaces are disabled.
```

File: python/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+configure_build_install_location(sdformat)
+
+if (BUILD_TESTING)
+  set(python_tests
+  )
+
+  foreach (test ${python_tests})
+    add_test(NAME ${test}.py COMMAND
+      "${PYTHON_EXECUTABLE}" "${CMAKE_SOURCE_DIR}/python/test/${test}.py")
```
Comment (azeey) [inline python/CMakeLists.txt:55]: Should we use `Python3_EXECUTABLE` instead? I think `PYTHON_EXECUTABLE` is 
deprecated and we now use `find_package(Python3)` in ign-cmake as of
https://github.com/ignitionrobotics/ign-cmake/pull/218

**Review by azeey — COMMENTED** [review]
Should we set the `cmake_minimum_required` version to 3.12 since `main` is 
targetting `focal` and later? This will ensure that we're using `find_package(Python3)` in ign-cmake.

**Review by azeey — APPROVED** [review]

**Pull request merged**